qiita_db/user.py

```diff
@@ -475,6 +477,46 @@ def unread_messages(self):
             return qdb.sql_connection.TRN.execute_fetchindex()
 
     # ------- methods ---------
+    def user_artifacts(self, artifact_type=None):
+        """Returns the artifacts owned by the user, grouped by study
+
+        Parameters
+        ----------
+        artifact_type : str, optional
+            The artifact type to retrieve. Default: retrieve all artfact types
+
+        Returns
+        -------
+        dict of {qiita_db.study.Study: list of qiita_db.artifact.Artifact}
+            The artifacts owned by the user
+        """
+        with qdb.sql_connection.TRN:
+            sql_args = [self.id, qiita_config.portal]
+            sql_a_type = ""
+            if artifact_type:
```

Comment:
I see your point. This has been added mainly for the needs in the interface. I don't think though that an empty dict is a bad answer. What do others think?